torch/csrc/jit/serialization/import.h

```diff
@@ -16,6 +17,29 @@ class ReadAdapterInterface;
 namespace torch {
 namespace jit {
 
+// used in torch.package deserialization
+class TORCH_API StorageContext {
```

Comment:
I doubt if we really need `StorageContext` here. Given that `root_key` is updated, the prefix can be added to the `root_key` to indicate the prefix. For the previous model, the `root_key` is the tensor index. After this format change, the `root_key` will be 
```
    (torch._utils._rebuild_tensor_v2(pers.obj(('storage',
          torch.DoubleStorage,
          '140671917317488.storage',
          'cpu',
          8),),
       0,
       (2, 4),
       (4, 1),
       False,
       collections.OrderedDict()),
```
What I suggest is, make it 
```
```
    (torch._utils._rebuild_tensor_v2(pers.obj(('storage',
          torch.DoubleStorage,
          '{archive_name}/140671917317488.storage',
          'cpu',
          8),),
       0,
       (2, 4),
       (4, 1),
       False,
       collections.OrderedDict()),
```
Then we don't need to add different *_prefix. To handle old models, the default search path can be it's own archive. Anyway, it's just my suggestion to make it clean